singer-io/tap-jira
# TDL-17703 Update projects stream

# Description of change
- Updated the endpoint for the projects stream as the one used in the tap was deprecated ([Ref Link](https://developer.atlassian.com/cloud/jira/platform/rest/v2/api-group-projects/#api-rest-api-2-project-get))
- As the new endpoint supported pagination, added pagination support for the sync
- Updated integration tests for paginated sync
- Added unittests for the code change.

# Manual QA steps
 - Run the sync and check whether the records are returned in a paginated manner.
 - Check the path is changed for projects stream.
 
# Risks
 - 
 
# Rollback steps
 - revert this branch
